sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer.py

```diff
@@ -0,0 +1,163 @@
+    TIMEOUT_SYMBOL,
+    RECEIVER_RUNTIME_METRIC_SYMBOL,
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EventHubConsumer(ConsumerProducerMixin):  # pylint:disable=too-many-instance-attributes
+    """
+    A consumer responsible for reading EventData from a specific Event Hub
+    partition and as a member of a specific consumer group.
+
+    A consumer may be exclusive, which asserts ownership over the partition for the consumer
+    group to ensure that only one consumer from that group is reading the from the partition.
+    These exclusive consumers are sometimes referred to as "Epoch Consumers."
+
+    A consumer may also be non-exclusive, allowing multiple consumers from the same consumer
+    group to be actively reading events from the partition.  These non-exclusive consumers are
+    sometimes referred to as "Non-Epoch Consumers."
+
+    Please use the method `create_consumer` on `EventHubClient` for creating `EventHubConsumer`.
+    """
+
+    def __init__(self, client, source, **kwargs):
+        """
+        Instantiate a consumer. EventHubConsumer should be instantiated by calling the `create_consumer` method
+        in EventHubClient.
+
+        :param client: The parent EventHubClient.
+        :type client: ~azure.eventhub.client.EventHubClient
+        :param source: The source EventHub from which to receive events.
+        :type source: str
+        :param prefetch: The number of events to prefetch from the service
+         for processing. Default is 300.
+        :type prefetch: int
+        :param owner_level: The priority of the exclusive consumer. An exclusive
+         consumer will be created if owner_level is set.
+        :type owner_level: int
+        :param track_last_enqueued_event_properties: Indicates whether or not the consumer should request information
+         on the last enqueued event on its associated partition, and track that information as events are received.
+         When information about the partition's last enqueued event is being tracked, each event received from the
+         Event Hubs service will carry metadata about the partition. This results in a small amount of additional
+         network bandwidth consumption that is generally a favorable trade-off when considered against periodically
+         making requests for partition properties using the Event Hub client.
+         It is set to `False` by default.
+        :type track_last_enqueued_event_properties: bool
+        """
+        event_position = kwargs.get("event_position", None)
+        prefetch = kwargs.get("prefetch", 300)
+        owner_level = kwargs.get("owner_level", None)
+        keep_alive = kwargs.get("keep_alive", None)
+        auto_reconnect = kwargs.get("auto_reconnect", True)
+        track_last_enqueued_event_properties = kwargs.get("track_last_enqueued_event_properties", False)
+
+        self.running = False
+        self.closed = False
+        self.stop = False  # used by event processor
```

Comment:
Nit:  running, closed, stop(ped?)

If the variable semantic really is an imperative (do the stop), maybe should_stop

Reply:
Given that this is internal only I will leave as-is for now - just in case by renaming this I miss something.